vmm/src/config.rs

```diff
@@ -195,14 +192,89 @@ impl<'a> NetConfig<'a> {
     }
 }
 
+pub struct RngConfig<'a> {
+    pub src: &'a Path,
+}
+
+impl<'a> RngConfig<'a> {
+    pub fn parse(rng: &'a str) -> Result<Self> {
+        Ok(RngConfig {
+            src: Path::new(rng),
+        })
+    }
+}
+
+pub struct FsConfig<'a> {
+    pub tag: &'a str,
+    pub sock: &'a Path,
+    pub num_queues: usize,
+    pub queue_size: u16,
+}
+
+impl<'a> FsConfig<'a> {
+    pub fn parse(fs: Option<&'a str>) -> Result<Option<Self>> {
+        if fs.is_none() {
+            return Ok(None);
+        }
+
+        // Split the parameters based on the comma delimiter
+        let params_list: Vec<&str> = fs.unwrap().split(',').collect();
+
+        let mut tag: &str = "";
+        let mut sock: &str = "";
+        let mut num_queues_str: &str = "";
+        let mut queue_size_str: &str = "";
+
+        for param in params_list.iter() {
+            if param.starts_with("tag=") {
+                tag = &param[4..];
+            } else if param.starts_with("sock=") {
+                sock = &param[5..];
+            } else if param.starts_with("num_queues=") {
+                num_queues_str = &param[11..];
+            } else if param.starts_with("queue_size=") {
+                queue_size_str = &param[11..];
+            }
+        }
+
+        let mut num_queues: usize = 1;
+        let mut queue_size: u16 = 1024;
+
+        if tag.is_empty() {
+            return Err(Error::ParseFsTagParam);
+        }
+        if sock.is_empty() {
+            return Err(Error::ParseFsSockParam);
+        }
+        if !num_queues_str.is_empty() {
+            num_queues = num_queues_str
+                .parse()
+                .map_err(Error::ParseFsNumQueuesParam)?;
+        }
+        if !queue_size_str.is_empty() {
+            queue_size = queue_size_str
+                .parse()
+                .map_err(Error::ParseFsQueueSizeParam)?;
+        }
+
+        Ok(Some(FsConfig {
+            tag,
+            sock: Path::new(sock),
+            num_queues,
+            queue_size,
+        }))
+    }
+}
+
 pub struct VmConfig<'a> {
     pub cpus: CpusConfig,
     pub memory: MemoryConfig,
     pub kernel: KernelConfig<'a>,
     pub cmdline: CmdlineConfig,
     pub disks: Vec<DiskConfig<'a>>,
-    pub rng: RngConfig<'a>,
     pub net: Option<NetConfig<'a>>,
+    pub rng: RngConfig<'a>,
+    pub fs: Option<FsConfig<'a>>,
```

Comment:
I think it would make sense to support multiple shared directories, so we should have a vector of `FsConfig` here if possible.